homeassistant/components/connector/config_flow.py

```diff
@@ -0,0 +1,65 @@
+"""Config flow for Connector integration."""
+from __future__ import annotations
+
+import logging
+
+from connectorlocal.connectorlocal import ConnectorHub
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, CONF_HOST
+
+from .const import DEFAULT_HUB_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_API_KEY): vol.All(str, vol.Length(min=16, max=16)),
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Connector."""
+
+    VERSION = 1
+
+    CONNECTION_CLASS = config_entries.CONN_CLASS_UNKNOWN
+
+    def __init__(self):
+        """Initialize the connector hub."""
+        self.host = None
+        self.key = None
+        self.errors = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if user_input is not None:
+            self.host = user_input[CONF_HOST].split("&")
```

Comment:
Why do we allow the user to input multiple hosts instead of creating one config entry per host?

Reply:
When communicating with HUB through UDP, it needs to occupy port 32100